src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/AbstractBolt.java

```diff
@@ -127,16 +127,30 @@ protected void dispatch(Tuple input) throws Exception {
 
     protected void handleLifeCycleEvent(LifecycleEvent event) {
         if (Signal.START.equals(event.getSignal())) {
-            active = true;
             emit(ZkStreams.ZK.toString(), currentTuple, new Values(event, commandContext));
+            try {
+                activate();
+            } finally {
+                active = true;
+            }
         } else if (Signal.SHUTDOWN.equals(event.getSignal())) {
-            active = false;
             emit(ZkStreams.ZK.toString(), currentTuple, new Values(event, commandContext));
+            try {
+                deactivate();
+            } finally {
+                active = false;
+            }
         } else {
             log.error("Unsupported signal received: {}", event.getSignal());
         }
     }
 
+    protected void activate() {
+    }
```

Comment:
add commend "no actions required" or something similar. Now it looks like someone has forgotten to write the method body.

Reply:
Added.